src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/SupplierAssessmentService.kt

```diff
@@ -46,6 +50,8 @@ class SupplierAssessmentService(
       appointmentTime,
       SUPPLIER_ASSESSMENT,
       createdByUser,
+      appointmentDeliveryType,
+      appointmentDeliveryAddress
     )
     supplierAssessment.appointments.add(appointment)
```

Comment:
_this does not block this PR, can be discussed after it's merged_

I'm a bit confused with the chain of `createOrUpdateAppointment` then `appointments.add` -- is it "create" and "add"?

I can see the appointments are a set, but when thinking about appointments, order seems important, so I'm not sure set is a fitting data structure. What do you think?

Reply:
1. This logic is not part of my PR.. but I agree that it seems a bit wrong to me... If you are updating an appointment you don't really want to add it again to the supplierAssessment. I'm assuming that if it's a duplicate appointment then it will be discarded (rather than throw some type of hibernate exception)..
I'll see if I can add a hibernate test for this.


2. Yes, you make a good point, I think appointments would be ideally ordered based on some time value. Maybe something to raise in next dev meeting.

Reply:
Just did a test -- it's fine because the Appointment class has a equals() on appointment.id